components/script/stylesheet_loader.rs

```diff
@@ -150,144 +267,70 @@ impl FetchResponseListener for StylesheetContext {
-        owner.set_origin_clean(self.origin_clean);
-        if owner.parser_inserted() {
-            document.decrement_script_blocking_stylesheet_count();
-        }
+        let element = self.element.root();
+        let document = self.document.root();
+        let mut is_css = metadata.content_type.clone().is_some_and(|content_type| {
+            let mime: Mime = content_type.into_inner().into();
+            mime.type_() == mime::TEXT && mime.subtype() == mime::CSS
+        }) || (
+            // Quirk: If the document has been set to quirks mode,
+            // has the same origin as the URL of the external resource,
+            // and the Content-Type metadata of the external resource
+            // is not a supported style sheet type, the user agent must
+            // instead assume it to be text/css.
+            // <https://html.spec.whatwg.org/multipage/#link-type-stylesheet>
+            document.quirks_mode() == QuirksMode::Quirks &&
+                document.origin().immutable().clone() == metadata.final_url.origin()
+        );
 
         // From <https://html.spec.whatwg.org/multipage/#link-type-stylesheet>:
-        // > A link element of this type is implicitly potentially render-blocking if the element
-        // > was created by its node document's parser.
-        if matches!(self.source, StylesheetContextSource::LinkElement { .. }) &&
-            owner.parser_inserted()
+        // > Quirk: If the document has been set to quirks mode, has the same origin as
+        // > the URL of the external resource, and the Content-Type metadata of the
+        // > external resource is not a supported style sheet type, the user agent must
+        // > instead assume it to be text/css.
+        if document.quirks_mode() == QuirksMode::Quirks &&
+            document.url().origin() == self.url.origin()
         {
-            document.decrement_render_blocking_element_count();
+            is_css = true;
         }
 
-        document.finish_load(LoadType::Stylesheet(self.url.clone()), CanGc::note());
+        self.unminify_css(metadata.final_url.clone());
+        if !is_css {
+            self.data.clear();
+        }
 
-        if let Some(any_failed) = owner.load_finished(successful) {
-            let event = if any_failed {
-                atom!("error")
-            } else {
-                atom!("load")
-            };
-            element
-                .upcast::<EventTarget>()
-                .fire_event(event, CanGc::note());
+        if self.is_for_a_previous_generation(&element) {
+            self.do_post_parse_tasks(successful, None);
+            return;
         }
+
+        let loader = match pref!(dom_parallel_css_parsing_enabled) {
+            true => {
+                ElementStylesheetLoader::Asynchronous(AsynchronousStylesheetLoader::new(&element))
+            },
+            false => ElementStylesheetLoader::Synchronous { element: &element },
+        };
```

Comment:
Same ultra-nit about matching booleans:

```suggestion
        let loader = if pref!(dom_parallel_css_parsing_enabled) {
            ElementStylesheetLoader::Asynchronous(AsynchronousStylesheetLoader::new(&element))
        } else {
            ElementStylesheetLoader::Synchronous { element: &element }
        };
```

Reply:
This makes a lot of sense. I'll make the change.